code/go/0chain.net/smartcontract/storagesc/allocation.go

```diff
@@ -1155,24 +1079,17 @@ func (sc *StorageSmartContract) updateAllocationRequestInternal(
 		request.OwnerID = t.ClientID
 	}
 
-	var clist *Allocations // client allocations list
-	if clist, err = sc.getAllocationsList(request.OwnerID, balances); err != nil {
-		return "", common.NewError("allocation_updating_failed",
-			"can't get client's allocations list: "+err.Error())
-	}
-
-	if !clist.has(request.ID) {
-		return "", common.NewErrorf("allocation_updating_failed",
-			"can't find allocation in client's allocations list: %s (%d)",
-			request.ID, len(clist.List))
-	}
-
 	var alloc *StorageAllocation
 	if alloc, err = sc.getAllocation(request.ID, balances); err != nil {
 		return "", common.NewError("allocation_updating_failed",
 			"can't get existing allocation: "+err.Error())
 	}
 
+	if request.OwnerID != alloc.Owner {
```

Comment:
We need to verify if this request is send by owner or authorized by owner, otherwise could fake request that set the OnwerID as the alloc.Owner. 

If the OnwerID is the same as t.ClientID, it should be fine as the transaction signature will ensure that the t.ClientID is correct. However, if the OwnerID is not t.ClientID, then we will need to check how it is set, and need to verify the owner.